### ISSUE TITLE
<egXML> in ODD to XHTML

### ISSUE THREAD

**Issue by ddenison-man** [TEIC/Stylesheets#553]
Until March 2022, a multi-line example of XML coding wrapped in &lt;egXML&gt; ... &lt;/egXML&gt; in the ODD file had every new line left-aligned when displayed in XHTML. Now, using TEI P5 16.0.0, such examples are displayed with only the first line at the left edge and all subsequent lines hugely indented by 20 spaces. Alex Jitianu of Oxygen tells me 'It looks like it's something introduced in the 7.53.0 which comes bundled into the 16.0.0. addon'. A simple example is given below, before and after. Please can this be corrected.
Thanks
David Denison

![image](https://user-images.githubusercontent.com/33696247/170952514-199cb38c-dcf8-4462-aad1-00ea1164ce24.png)
![image](https://user-images.githubusercontent.com/33696247/170952583-d1003e7a-1db0-437c-8cb4-e0575bd0035c.png)

**Comment by sydb** [comment]
@ddenison-man — Can you post the ODD source to the snippet above? Is it an `<egXML>` child of a `<elementSpec>` (or similar), or inside a paragraph or some such?
Thanks.

**Comment by sydb** [comment]
Tests so far:

- Run against 7.52.0 with commandline roma: no problem
- Run against released with commandline roma: no problem
- Run with oXygen (using latest plug-in, I think): PROBLEM
- Run via 7.52.0/bin/teitohtml: no problem
- Run via released/bin/teitohtml: PROBLEM

I must admit, I find that combination somewhat confusing. In any case, where it says PROBLEM I found the same ugly indentation @ddenison-man reports in `<egXML>` when a child of `<p>` in the prose or in `schemaSpec/elementSpec/exemplum`.

@ddenison-man — you can ignore my query, above. This ticket will probably be assigned this coming Wed 01 Jun.

**Comment by ddenison-man** [comment]
It's just <egXML> as a child of <div> following a <p> in the discursive part of the schema, but it makes no difference whether the example is moved inside <p> or not:

                  <p>For direct literary or biblical quotations, the closing tag of the &lt;quote&gt; element is followed directly (without a space) by a &lt;bibl&gt; (bibliographic citation) element, with &lt;quote&gt; + &lt;bibl&gt; together wrapped in a &lt;cit&gt; (citation) element:</p>
<egXML xmlns="http://www.tei-c.org/ns/Examples">&lt;lb/&gt;[...] &lt;seg type="rp"&gt;that beautiful line of &lt;persName ref="psn:tbc"&gt;Pope&lt;/persName&gt;, where he describes
&lt;lb/&gt;the Indian who &lt;cit&gt;&lt;quote&gt;“sees God in Clouds and hears him
&lt;lb/&gt;in the wind”&lt;/quote&gt;&lt;bibl&gt;Alexander Pope, &lt;hi rend="italic"&gt;An Essay on Man,
Epistle I&lt;/hi&gt;.&lt;/bibl&gt;&lt;/cit&gt;&lt;/seg&gt;</egXML>

DD
--

<><><><><><><><><><><><><><><><><><><><><><><><><>
        Professor David Denison, FBA
        Professor Emeritus of English Linguistics
        School of Arts, Languages and Cultures
        University of Manchester | Manchester M13 9PL | U.K.
https://www.research.manchester.ac.uk/portal/david.denison.html

________________________________________

**Comment by sydb** [comment]
I.e., the following, right?
~~~xml
<p>For direct literary or biblical quotations, the closing tag of the <gi>quote</gi> element is followed directly (without a space)
  by a <gi>bibl</gi> (bibliographic citation) element, with <gi>quote</gi> + <gi>bibl</gi> together wrapped in a <gi>cit</gi>
  (citation) element:</p>

<egXML xmlns="http://www.tei-c.org/ns/Examples"><lb/>[...] <seg type="rp">that beautiful line of <persName ref="psn:tbc">Pope</persName>, where he describes
<lb/>the Indian who <cit><quote>“sees God in Clouds and hears him
<lb/>in the wind”</quote><bibl>Alexander Pope, <hi rend="italic">An Essay on Man,
Epistle I</hi>.</bibl></cit></seg></egXML>
~~~
I have created a [tiny ODD file](https://github.com/TEIC/Stylesheets/files/8800520/DDT.odd.txt) for testing this. Note that I have appended a “.txt” extension so that GitHub will not think it a malicious file.

**Comment by ddenison-man** [comment]
Dear Syd (if I may),

Your test file exhibits exactly the same behaviour as my long schema, giving the ugly indents from l.2 onwards with the latest stylesheets, but displaying fine with the previous stylesheets.

I apologise for giving the ODD source of a different snippet, an earlier selection that was messier than the one I sent screenshots of to this list. As far as I can see, you've got the code right.

Thanks for the speedy response.

best
David

--

<><><><><><><><><><><><><><><><><><><><><><><><><>
        Professor David Denison, FBA
        Professor Emeritus of English Linguistics
        School of Arts, Languages and Cultures
        University of Manchester | Manchester M13 9PL | U.K.
https://www.research.manchester.ac.uk/portal/david.denison.html

________________________________________

**Comment by sydb** [comment]
No worries! I hope we can fix this before the next release, it is really quite ugly.

**Comment by sydb** [comment]
@bleekere and I assigned this to @sabineseifert (and @peterstadler:-) for the task of finding out what changed in the code between last release and now that caused this problem.

**Comment by peterstadler** [comment]
Just did some debugging and it seems to me the root cause (i.e. the major change from v7.52.0 to v7.53.0) is the changed HTML output serialization. If I remember correctly, we decided to align html5 and html outputs (setting html5 as default) and in this course various `xsl:output` elements have been updated, including the addition of `indent="yes"`, see e.g. 89f1a9d.

@HelenaSabel  I don't know whether the introduction of the `@indent` attribute was simply to beautify the output? If so, we could simply remove it (the `indent="yes"` attribute) again (or explicitly set it to "no") to fix the issue at hand.
